src/hooks/server_changed.rs

```diff
@@ -296,77 +295,74 @@ fn expand_volume(peers: Vec<Peer>, volume_info: Option<Volume>) -> Result<i32, S
 fn get_brick_list(peers: &Vec<Peer>,
                   volume: Option<Volume>)
                   -> Result<Vec<gluster::volume::Brick>, Status> {
-
     // Default to 3 replicas if the parsing fails
+    let mut brick_devices: Vec<block::BrickDevice> = Vec::new();
+
     let replica_config = get_config_value("replication_level").unwrap_or("3".to_string());
     let replicas = replica_config.parse().unwrap_or(3);
-    let config_brick_devices: Vec<String> = get_config_value("brick_devices")
-        .unwrap_or("".to_string())
-        .split(" ")
-        .map(|s| s.to_string())
-        .filter(|s| !s.is_empty())
-        .collect();
-    log!(format!("user config storage_list: {:?}", config_brick_devices));
-    let mut brick_paths: Vec<String> = Vec::new();
 
-    let juju_storage_bricks = juju::storage_list().unwrap();
-    log!(format!("juju storage_list: {:?}", juju_storage_bricks));
-
-    // Add any devices that were set with juju storage
-    for brick in juju_storage_bricks.lines() {
-        // This is the /dev/ location.
-        let storage_location = juju::storage_get(brick.trim()).unwrap();
-        // Translate to mount location
-        let brick_path = PathBuf::from(storage_location.trim());
-        let brick_filename = match brick_path.file_name() {
-            Some(name) => name,
-            None => {
-                log!(format!("Unable to determine filename for device: {:?}. Skipping",
-                             brick),
-                     Error);
-                continue;
-            }
-        };
-        let mount_path = format!("/mnt/{}", brick_filename.to_string_lossy());
+    // TODO: Should this fail the hook or just keep going?
+    log!("Checking for ephemeral unmount");
+    ephemeral_unmount().map_err(|e| Status::InvalidConfig(e))?;
 
-        brick_paths.push(mount_path);
-    }
+    // Get user configured storage devices
+    let manual_brick_devices = block::get_manual_bricks().map_err(|e| Status::InvalidConfig(e))?;
+    brick_devices.extend(manual_brick_devices);
 
-    // Add any devices that were set with the config.yaml
-    for brick in config_brick_devices {
-        let brick_path = PathBuf::from(brick);
-        log!(format!("brick_path: {:?}", brick_path));
+    // Get the juju storage block devices
+    let juju_config_brick_devices = block::get_juju_bricks().map_err(|e| Status::InvalidConfig(e))?;
+    brick_devices.extend(juju_config_brick_devices);
```

Comment:
This is the second place I've seen this code to collect all of the block devices, would a shared helper function help?